crates/common/src/header.rs

```diff
@@ -24,14 +24,26 @@ pub struct BlockHeader {
     pub l1_da_mode: L1DataAvailabilityMode,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Default, Dummy, serde::Serialize, serde::Deserialize)]
+#[derive(
+    Debug, Copy, Clone, PartialEq, Eq, Default, Dummy, serde::Serialize, serde::Deserialize,
+)]
 #[serde(rename_all = "UPPERCASE")]
 pub enum L1DataAvailabilityMode {
     #[default]
     Calldata,
     Blob,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Default, Dummy)]
+pub struct PendingBlockHeader {
```

Comment:
Why do we need this extra type?

Reply:
We don't at this stage. I was planning on changing the gateway pending block to return this instead. To more accurately model things.

This was used by the previous commit; at this point the PR draft contains quite a few unused items.

Reply:
Will remove the type in the actual PR. The gateway pending thing can be a separate PR if needed.